src/storage-types/src/connections.rs

```diff
@@ -1420,16 +1415,50 @@ impl MySqlConnection<InlinedConnection> {
             opts = opts.pass(Some(password));
         }
 
-        // TODO(roshan): Implement SSL mode support
-        if self.tls_mode != MySqlSslMode::Disabled {
-            return Err(anyhow!("MySQL TLS modes are not yet supported"));
+        // Our `MySqlSslMode` enum matches the official MySQL Client `--ssl-mode` parameter values
+        // which uses opt-in security features (SSL, CA verification, & Identity verification).
+        // The mysql_async crate `SslOpts` struct uses an opt-out mechanism for each of these, so
+        // we need to appropriately disable features to match the intent of each enum value.
```

Comment:
Oh but `tokio-postgres` exposes this enum itself, right?

Reply:
yep `tokio-postgres` has an enum that matches the postgresql client option: https://docs.rs/tokio-postgres/latest/tokio_postgres/config/enum.SslMode.html